cypress/support/commands/toolbar.js

```diff
@@ -1,36 +1,42 @@
 /* eslint-disable no-undef */
 
 // toolbarButton: String for the text of the toolbar button to click.
-// dropdownButton: String for the text of the dropdown button to click after the toolbar dropdown is opened.
-Cypress.Commands.add('toolbar', (toolbarButton, dropdownButton = '') => {
-  cy.get('#toolbar').get('button').then((toolbarButtons) => {
-    const nums = [...Array(toolbarButtons.length).keys()];
-    nums.forEach((index) => {
-      const button = toolbarButtons[index].children[0];
-      if (button && button.innerText && button.innerText.includes(toolbarButton)) {
-        button.click();
-        return;
+// toolbarOption: String for the text of the dropdown button to click after the toolbar dropdown is opened.
+Cypress.Commands.add('toolbar', (toolbarButton, toolbarOption = '') => {
+  const clickToolbarButton = cy
+    .get('#toolbar')
+    .find('button')
+    .then((buttons) => {
+      const targetToolbarButton = [...buttons].find(
+        (btn) => btn.innerText.trim() === toolbarButton
+      );
+
+      if (!targetToolbarButton) {
+        cy.logAndThrowError(`Toolbar button: "${toolbarButton}" was not found`);
       }
+      return cy.wrap(targetToolbarButton).click();
     });
-  });
 
-  if (dropdownButton) {
-    return cy.get('.bx--overflow-menu-options').then((dropdownButtons) => {
-      const buttons = dropdownButtons.children();
-      for (let index = 0; index < buttons.length; index++) {
-        const button = buttons[index];
-        if (
-          button &&
-          button.innerText &&
-          button.innerText.includes(dropdownButton)
-        ) {
-          return cy.wrap(button.children[0]).click();
+  // If toolbarOption is provided, wait for toolbar to open,
+  // then look for the given toolbar option
+  if (toolbarOption) {
+    return clickToolbarButton.then(() => {
+      return cy.get('.bx--overflow-menu-options li').then((toolbarOptions) => {
+        const targetToolbarOption = [...toolbarOptions].find(
+          (option) => option.innerText.trim() === toolbarOption
+        );
+
+        if (!targetToolbarOption) {
+          cy.logAndThrowError(`"${toolbarOption}" option was not found in the "${toolbarButton}" toolbar`);
         }
-      }
-      return cy.wrap(null);
+        // returning the cypress chainable to the top of the command scope
+        return cy.wrap(targetToolbarOption).click();
```

Comment:
I like how we're returning the chainable object or throwing an error so it looks like you can chain it without concern about it not being found.